Changes to loader parser broke not including -loader parameter

https://github.com/FabricMC/fabric-installer/pull/150

It seems this now actually broke not passing -loader in. Previously, this did at least function. Now, it fully errors.
This seems to violate the help as well still, which implies it will be "latest".


java -jar fabric-installer.jar server -mcversion 1.21.1 -downloadMinecraft -noprofile


Loading Fabric Installer: 1.0.2
Exception in thread "main" java.lang.RuntimeException: Failed to install Server
    at net.fabricmc.installer.Main.main(Main.java:89)
Caused by: java.lang.IllegalArgumentException: Could not find loader in the arguments
    at net.fabricmc.installer.util.ArgumentParser.get(ArgumentParser.java:38)
    at net.fabricmc.installer.Handler.getLoaderVersion(Handler.java:297)
    at net.fabricmc.installer.server.ServerHandler.installCli(ServerHandler.java:76)
    at net.fabricmc.installer.Main.main(Main.java:87)

--- a/src/main/java/net/fabricmc/installer/Main.java
+++ b/src/main/java/net/fabricmc/installer/Main.java
@@ -50,8 +50,8 @@ public class Main {
 		String command = argumentParser.getCommand().orElse(null);
 
 		//Can be used if you wish to re-host or provide custom versions. Ensure you include the trailing /
-		String metaUrl = argumentParser.has("metaurl") ? argumentParser.get("metaurl") : null;
-		String mavenUrl = argumentParser.has("mavenurl") ? argumentParser.get("mavenurl") : null;
+		String metaUrl = argumentParser.get("metaurl");
+		String mavenUrl = argumentParser.get("mavenurl");
 
 		if (metaUrl != null || mavenUrl != null) {
 			FabricService.setFixed(metaUrl, mavenUrl);

--- a/src/main/java/net/fabricmc/installer/util/ArgumentParser.java
+++ b/src/main/java/net/fabricmc/installer/util/ArgumentParser.java
@@ -19,11 +19,10 @@ package net.fabricmc.installer.util;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
-import java.util.function.Consumer;
 import java.util.function.Supplier;
 
 public class ArgumentParser {
-	private String[] args;
+	private final String[] args;
 	private Map<String, String> argMap;
 	//The command will be the first argument passed, and if it doesnt start with -
 	private String command = null;
@@ -34,35 +33,21 @@ public class ArgumentParser {
 	}
 
 	public String get(String argument) {
-		if (!argMap.containsKey(argument)) {
-			throw new IllegalArgumentException(String.format("Could not find %s in the arguments", argument));
-		}
-
-		String arg = argMap.get(argument);
-
-		if (arg == null) {
-			throw new IllegalArgumentException(String.format("Could not value for %s", argument));
-		}
-
-		return arg;
+		return argMap.get(argument);
 	}
 
 	public String getOrDefault(String argument, Supplier<String> stringSuppler) {
-		if (!argMap.containsKey(argument)) {
-			return stringSuppler.get();
+		String ret = argMap.get(argument);
+
+		if (ret == null) {
+			ret = stringSuppler.get();
 		}
 
-		return argMap.get(argument);
+		return ret;
 	}
 
 	public boolean has(String argument) {
 		return argMap.containsKey(argument);
-	}
-
-	public void ifPresent(String argument, Consumer<String> consumer) {
-		if (has(argument)) {
-			consumer.accept(get(argument));
-		}
 	}
 
 	public Optional<String> getCommand() {

--- a/src/test/java/net/fabricmc/installer/test/ArgumentTests.java
+++ b/src/test/java/net/fabricmc/installer/test/ArgumentTests.java
@@ -43,14 +43,14 @@ public class ArgumentTests {
 		ArgumentParser.create(new String[]{"-arg1", "Hello", "-arg1"});
 	}
 
-	@Test(expected = IllegalArgumentException.class)
+	@Test
 	public void testUnknownArg() {
-		ArgumentParser.create(new String[]{"-arg1", "Hello"}).get("arg2");
+		Assert.assertNull(ArgumentParser.create(new String[]{"-arg1", "Hello"}).get("arg2"));
 	}
 
-	@Test(expected = IllegalArgumentException.class)
+	@Test
 	public void testNullArg() {
-		ArgumentParser.create(new String[]{"-arg1"}).get("arg1");
+		Assert.assertNull(ArgumentParser.create(new String[]{"-arg1"}).get("arg1"));
 	}
 
 	@Test

